src/bindings/jna/libelektra/src/main/java/org/libelektra/Key.java

```diff
@@ -231,6 +311,24 @@ public boolean getBoolean ()
 		return Boolean.parseBoolean (getString ());
 	}
 
+	/**
+	 * Return the key's value and releases this key
+	 *
+	 * @return {@link #getString()} interpreted as boolean value
+	 * @throws KeyBinaryTypeNotSupportedException if the underlying native key is of
+	 *                                            type binary
+	 * @throws KeyReleasedException               if this {@link Key} has already
+	 *                                            been released
+	 * @apiNote This convenience method is primarily intended for facilityting early
+	 *          clean up in the case when a `Key` obtained via any of its public
+	 *          methods or the public methods of {@link KeySet} is just being read
+	 *          once and is not needed afterwards.
+	 */
+	public boolean getBooleanAndRelease ()
```

Comment:
I basically agree, but would like to debate the removal once the automated clean-up works. Nonetheless, depending on the use-case, manual release could prove a worthwhile optimization. (e.g. working with a potentially large amount of keys within the same scope)

Reply:
I'm certainly no expert on this, but as far as I understand it, the JVM GC (at least partially) runs in a separate thread and could clean-up objects more or less at any point. I think there is also some logic to quickly dispose of temporary objects, since in Java it is quite common have extremely short-lived objects.

Reply:
Addendum: If the add these functions, I think we should at least have specific use-case where they make sense, because I think they might actually hurt performance in many cases. `Cleaner` clearly states that it runs in a separate thread. However, when using `getBooleanAndRelease()` the `keyDel`/`ksDel` call is performed in a blocking manor.

So doing this
```java
for (String s : hugeList) {
    foo (ks.lookup (s).map(k -> k.getBooleanAndRelease()));
}
```

Might (*) reduce the maximum RAM use compared to
```java
for (String s : hugeList) {
    foo (ks.lookup (s).map(k -> k.getBoolean()));
}
```

But it is probably slower, because we need to wait for the key to be deleted before calling `foo` and moving on to the next key.

(*) It's not certain, because of parallel GC. I'd even say it is unlikely there is any benefit, because the JVM will probably trigger the a GC run when RAM use rises and all the keys that would be deleted by `*AndRelease()` aren't reachable outside the lambda so the `Cleaner` will delete them quickly.

Reply:
Thank you for your input! I think once we got the Cleaner to work without race conditions, we should only keep *AndRelease functions if we do a benchmark confirming a use case where it actually yields a significant benefit of using them.

Reply:
see #3874